### ISSUE TITLE
Look into using testResultsProcessor for real-time reporting

### ISSUE THREAD

**Issue by orta** [orta/vscode-jest#160]
### Feature

Right now the tool works by Jest doing a `--watch` run which exports JSON, which is pulled in by the extension and parsed for interesting data, switching that pattern to use the setting `testResultsProcessor` would allow for incremental updates and less file system faffing.

re: https://github.com/facebook/jest/issues/4806

**Comment by seanpoulter** [comment]
Here's the link to the [relevant Jest documentation](https://facebook.github.io/jest/docs/en/configuration.html#testresultsprocessor-string).

**Comment by seanpoulter** [comment]
Using the `testResultsProcessor` calls a node module from Jest. Any ideas how to get that back into VS Code @orta?

**Comment by orta** [comment]
I think it would need to open up an IPC connection to the running extension, I wonder if that's code that can live in jest-editor-support, because that could be useful in many places.

**Comment by seanpoulter** [comment]
The `testResultsProcessor` is called just before the JSON is output ([code pointer](https://github.com/facebook/jest/blob/6f7c85a29886afe0041ba74c7856cf54893d80c6/packages/jest-cli/src/run_jest.js#L56))

I'd suggest we try a custom reporter to stream results before the full run is done using [`--reporters`](https://facebook.github.io/jest/docs/en/configuration.html#reporters-array-modulename-modulename-options).

**Comment by orta** [comment]
Yeah, I wonder if a lang server process could be based entirely on that callback alone
